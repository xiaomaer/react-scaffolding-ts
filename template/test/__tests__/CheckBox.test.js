import React from 'react';
import Checkbox from '../../src/components/CheckBox/index';

// 测试组件的渲染
test('render checkbox', () => {
    const wrapper = shallow(
        <Checkbox id="xiao" name="xxx" value="xxx">
            label
        </Checkbox>
    );
    expect(wrapper).toMatchSnapshot();
});

// 测试组件props
test('render a checked props', () => {
    const wrapper = shallow(
        <Checkbox className="xm" id="xiao" name="xxx" value="xxx">
            label
        </Checkbox>
    );
    expect(wrapper.prop('className')).toEqual('xm');
});

// 测试组件事件处理
test('test onchange event', () => {
    const wrapper = shallow(<Checkbox value="xxx">label</Checkbox>);
    expect(wrapper).toMatchSnapshot();
    wrapper.find('input').simulate('change', {
        target: {
            checked: true
        }
    });
    expect(wrapper).toMatchSnapshot();
});
