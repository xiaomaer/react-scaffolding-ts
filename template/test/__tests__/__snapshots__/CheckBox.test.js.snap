// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render checkbox 1`] = `
<div
  className=""
  style={Object {}}
>
  <input
    checked={false}
    defaultChecked={false}
    id="xiao"
    name="xxx"
    onChange={[Function]}
    type="checkbox"
    value="xxx"
  />
  <span>
    label
  </span>
</div>
`;

exports[`test onchange event 1`] = `
<div
  className=""
  style={Object {}}
>
  <input
    checked={false}
    defaultChecked={false}
    onChange={[Function]}
    type="checkbox"
    value="xxx"
  />
  <span>
    label
  </span>
</div>
`;

exports[`test onchange event 2`] = `
<div
  className=""
  style={Object {}}
>
  <input
    checked={true}
    defaultChecked={false}
    onChange={[Function]}
    type="checkbox"
    value="xxx"
  />
  <span>
    label
  </span>
</div>
`;
